vm/devices/storage/disk_nvme/nvme_driver/fuzz/fuzz_nvme_driver.rs

```diff
@@ -0,0 +1,184 @@
+        let mem = DeviceSharedMemory::new(base_len, payload_len);
+
+        // Trasfer buffer
+        let payload_mem = mem
+            .guest_memory()
+            .subrange(base_len as u64, payload_len  as u64, false)
+            .unwrap();
+
+        // Nvme device and driver setup
+        let driver_source =
+            VmTaskDriverSource::new(SingleDriverBackend::new(driver));
+        let mut msi_set = MsiInterruptSet::new();
+        let nvme = NvmeController::new(
+            &driver_source,
+            mem.guest_memory().clone(),
+            &mut msi_set,
+            &mut ExternallyManagedMmioIntercepts,
+            NvmeControllerCaps {
+                msix_count: 2,
+                max_io_queues: 64,
+                subsystem_id: Guid::new_random(),
+            },
+        );
+
+        // Create nvme namespace
+        nvme.client()
+            .add_namespace(1, Arc::new(ram_disk))
+            .await
+            .unwrap();
+
+        let device = FuzzEmulatedDevice::new(nvme, msi_set, mem);
+        let nvme_driver = NvmeDriver::new(&driver_source, 64, device).await.unwrap();
+        let namespace = nvme_driver.namespace(1).await.unwrap();
+
+        Self {
+            driver: Some(nvme_driver),
+            namespace,
+            payload_mem,
+        }
+    }
+
+    /// Clean up fuzzing infrastructure.
+    pub async fn shutdown(&mut self) {
+        self.namespace
+            .deallocate(
+                0,
+                &[
+                    DsmRange {
+                        context_attributes: 0,
+                        starting_lba: 1000,
+                        lba_count: 2000,
+                    },
+                    DsmRange {
+                        context_attributes: 0,
+                        starting_lba: 2,
+                        lba_count: 2,
+                    },
+                ],
+            )
+            .await
+            .unwrap();
+
+        self.driver.take().unwrap().shutdown().await;
+    }
+
+    /// Generates and returns an aribtrary NvmeDriverAction
+    /// Returns a NotEnoughData error if an action was not generated, caller must handle this.
+    pub fn get_arbitrary_action(&self) -> arbitrary::Result<NvmeDriverAction>{
+        // Get required amount of arbitrary bytes
+        let arbitrary_data = get_raw_data(size_of::<NvmeDriverAction>());
+
+        match arbitrary_data {
+            Ok(data) => {
+                let mut u = Unstructured::new(&data);
+
+                // Generate arbitrary action
+                let action = u.arbitrary()?;
+                return Ok(action);
+            },
+            Err(e) => {
+               // Bubble up errors
+               return Err(e);
+            }
+        }
+    }
+
+    /// Executes a NvmeDriverAction on the nvme driver.
+    pub async fn execute_action(&mut self, action: NvmeDriverAction) {
+        match action {
+            NvmeDriverAction::Read { lba, block_count, target_cpu} => {
```

Comment:
Hm, it may be more prudent just to fuzz `disk_nvme`. For example, I ran this locally and hit an assert (`block_count <= self.max_transfer_block_count`). That's a righteous assert. So either we need to signal that "yep, that's expected" or we need to not violate those contracts.

I think we're going to need to work through this from both angles (I bet the nvme driver panics when receiving bad input from the host), but I'm interested in your opinion here ...
1) change what we fuzz to avoid panics, and/or
2) have a way to identify panics as "expected"

Reply:
It runs :)

```
$> cargo xtask fuzz run fuzz_nvme_driver
...
thread '<unnamed>' panicked at /home/mattkur/openvmm/vm/devices/storage/disk_nvme/nvme_driver/src/namespace.rs:259:9:
assertion failed: block_count <= self.max_transfer_block_count
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Reply:
Can we author a new `panic_bad_fuzzer` that detects if we're fuzzing? If so, bail out execution to the `do_fuzz` loop (which would reject the corpus?). And, if not fuzzing, just `panic`?

Reply:
^ Already discussed above. Will be part of the git issue and tracked there.